cli/src/main/scala/com/advancedtelematic/tuf/cli/Executor.scala

```diff
@@ -0,0 +1,164 @@
+package com.advancedtelematic.tuf.cli
+
+import java.nio.file.Files
+
+import com.advancedtelematic.libtuf.data.ClientDataType.RootRole
+import com.advancedtelematic.libtuf.reposerver.{UserDirectorHttpClient, UserReposerverClient, UserTufServerClient}
+import com.advancedtelematic.tuf.cli.Commands._
+import com.advancedtelematic.tuf.cli.DataType.{Director, RepoServer}
+import com.advancedtelematic.tuf.cli.repo.{DirectorRepo, RepoManagement, RepoServerRepo, TufRepo}
+import org.slf4j.LoggerFactory
+import TryToFuture._
+import io.circe.syntax._
+import com.advancedtelematic.libtuf.data.TufCodecs._
+import com.advancedtelematic.libtuf.data.ClientCodecs._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+case class MissingTargetKeyname() extends Exception
+
+abstract class Executor[C <: UserTufServerClient, R <: TufRepo](config: Config, repoServer: Future[C], tufRepo: R)
```

Comment:
I think I don't understand something. Isn't the logic to process the commands the same for reposerver or director? What changes is where the http requests go, as well as the local filesystem organization, but the command processing should be the same? The logic just somehow be contained in each Repo class ?

For example, moving offline should be the same from the command perspective, it just calls `repo.moveOffline`, but `repo` is an instance of a different class depending on director or repo ?

Reply:
`RepoServerRepo.moveRootOffline` and `DirectorRepo.moveRootOffline` are slightly different in that only the former has the `newTargetsName` parameter. I could have ignored the additional param in the new method, but I prefer the typesafyness of leaving it out.

Reply:
But this is a lot of code to go around ignoring one parameter. 

There is also a lot of checking for repo type, I think this should be kind of encapsulated in one place, and the natural way would be for it to be inside difference TufRepo classes. I am not really getting the need for this executor thing, or at least for multiple classes of an executor.

Reply:
`UserDirectorHttpClient` doesn't know anything about targets, so only `UserReposerverHttpClient` has methods dealing with them, so the `ReposerverExecutor` has to reference the type `UserReposerverHttpClient` explicitly.

Reply:
But that client could also be abstract and just throw an error when you are trying to get targets from a repository that doesn't support getting targets?